src/Mvc/Mvc.Core/src/Authorization/AuthorizeFilter.cs

```diff
@@ -140,14 +152,13 @@ private async Task<AuthorizationPolicy> GetEffectivePolicyAsync(AuthorizationFil
                     
                     if (context.Filters[i] is AuthorizeFilter authorizeFilter)
                     {
-                        builder = builder ?? new AuthorizationPolicyBuilder(effectivePolicy);
-                        builder.Combine(authorizeFilter.Policy);
```

Comment:
This logic is a little bit different right? `.Policy` is different stuff than `ComputePolicyAsync`

Reply:
I guess this is the same as what's in 2.2/3.0. Is the difference between these something to be concerned about?

Reply:
Yeah 2.1 was different from 2.2/3.0 to start, I am trying to make all of them the same with this PR

Reply:
I get that this changes the code to be more consistent, but does this change the observable behavior?

Reply:
So maybe we shouldn't port this change to 2.1, it looks like we never ported this 2.2 fix to 2.1 which is why the code bases are so different...  https://github.com/aspnet/Mvc/commit/a5083d525b76ed8699d2b94e91d7ef52106d65aa#diff-fbf649a01b0521186bc5d75f07c20d8f